src/main/kotlin/com/exactpro/th2/common/message/MessageUtils.kt

```diff
@@ -116,3 +122,33 @@ fun Date.toTimestamp(): Timestamp = toInstant().toTimestamp()
 fun LocalDateTime.toTimestamp(zone: ZoneOffset) : Timestamp = toInstant(zone).toTimestamp()
 fun LocalDateTime.toTimestamp() : Timestamp = toTimestamp(ZoneOffset.of(TimeZone.getDefault().id))
 fun Calendar.toTimestamp() : Timestamp = toInstant().toTimestamp()
+
+fun Message.toTreeTable(): TreeTable = TreeTableBuilder().apply {
+    for ((key, value) in fieldsMap) {
+        row(key, value.toTreeTableEntry())
+    }
+}.build()
+
+private fun Value.toTreeTableEntry(): TreeTableEntry {
+    if (hasMessageValue()) {
+        val nestedMessageValue = messageValue
+        val collectionBuilder = CollectionBuilder()
+        for ((key, value) in nestedMessageValue.fieldsMap) {
+            collectionBuilder.row(key, value.toTreeTableEntry())
+        }
+        return collectionBuilder.build()
+    }
+    if (hasListValue()) {
+        val collectionBuilder = CollectionBuilder()
+        for ((index, nestedValue) in listValue.valuesList.withIndex()) {
+            val nestedName = index.toString()
+            collectionBuilder.row(nestedName, nestedValue.toTreeTableEntry())
+        }
+        return collectionBuilder.build()
+    }
+    return RowBuilder()
+        .column(MessageTableColumn(simpleValue))
+        .build()
+}
+
+internal data class MessageTableColumn(val fieldValue: String) : IColumn
```

Comment:
I am not sure that Jackson will be able to serialize that object normally because it is a package-private. Could you please check it?